### ISSUE TITLE
Bad pip installations?  Example of matter client connections

### ISSUE THREAD

**Issue by bkoblenz** [home-assistant-libs/python-matter-server#1002]
Apologies if this is the wrong forum.  I am relatively new to matter and am trying to create a hub to integrate matter sensors with other sensors I already work with and have an existing UI and application.

I built a fresh Ubuntu (noble) installation and have tried to get the python-matter-server running.  I finally did get something to run by installing:
pip install home-assistant-chip-core
pip install home-assistant-chip-clusters
pip install home-assistant-chip-repl
pip install cryptography
pip install python-matter-server
pip install zeroconf
pip install websockets
mkdir /data
sudo chown pi:pi /data

That list got built up as
$ python -m matter_server.server
kept exposing issues.  In fact, I had to comment out a line from client/models/device_types.py that referenced (what I assume to be stale)
all_clusters.BarrierControl.

Looking at that file here the BarrierControl reference is already deleted so I think the sources I am downloading are inconsistent.
Should I throw away the pip installed stuff and grab the source tree from here?

I do have the server running but got the following network message in the log  (I do not have my borderrouter set up yet but what network might they be referring too?):
2024-12-15 14:14:32.828 (MainThread) INFO [chip.FabricAdmin] Allocating new controller with CaIndex: 1, FabricId: 0x0000000000000001, NodeId: 0x000000000001B669, CatTags: []
2024-12-15 14:14:33.158 (Dummy-2) CHIP_ERROR [chip.native.DIS] Failed to advertise records: src/inet/UDPEndPointImplSockets.cpp:421: OS Error 0x02000065: Network is unreachable
2024-12-15 14:14:33.169 (Dummy-2) CHIP_ERROR [chip.native.DL] Long dispatch time: 334 ms, for event type 2
2024-12-15 14:14:33.210 (MainThread) INFO [matter_server.server.vendor_info] Loading vendor info from storage

With the server running (I can see it listening on port 5580) I cannot get a client connection (aiohttp_session seems missing??.  This makes we wonder if my installation is missing something critical.  Is MatterClient the preferred way to connect to the server?  Any insight?

>>> from matter_server.client.client import MatterClient
>>> async def main():
...     async with MatterClient("localhost", 5580) as client:
...         nodes = await client.get_nodes()
...         print('nodes',nodes)
...
>>> asyncio.run(main())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "<stdin>", line 2, in main
  File "/home/pi/.venv/general/lib/python3.12/site-packages/matter_server/client/client.py", line 791, in __aenter__
    await self.connect()
  File "/home/pi/.venv/general/lib/python3.12/site-packages/matter_server/client/client.py", line 616, in connect
    await self.connection.connect()
  File "/home/pi/.venv/general/lib/python3.12/site-packages/matter_server/client/connection.py", line 67, in connect
    self._ws_client = await self._aiohttp_session.ws_connect(
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'int' object has no attribute 'ws_connect'

**Comment by bkoblenz** [comment]
I downloaded these sources and found (very helpful) the scripts/example.py) and tried to run it.

It also seemed to have a problem with the network (what network is the issue?).  Are there certain parameters required to run this example?

(general) pi@raspberrypi:~/python-matter-server/scripts$ python example.py
...
2024-12-15 15:20:13 raspberrypi chip.storage[2609] INFO Initializing persistent storage from file: /home/pi/.matter_server/chip.json
2024-12-15 15:20:13 raspberrypi chip.storage[2609] INFO Loading configuration from /home/pi/.matter_server/chip.json...
2024-12-15 15:20:13 raspberrypi chip.CertificateAuthority[2609] INFO Loading certificate authorities from storage...
2024-12-15 15:20:13 raspberrypi chip.CertificateAuthority[2609] INFO New CertificateAuthority at index 1
2024-12-15 15:20:13 raspberrypi chip.CertificateAuthority[2609] INFO Loading fabric admins from storage...
2024-12-15 15:20:13 raspberrypi chip.FabricAdmin[2609] INFO New FabricAdmin: FabricId: 0x0000000000000001, VendorId = 0xFFF1
2024-12-15 15:20:13 raspberrypi matter_server.server.stack[2609] INFO CHIP Controller Stack initialized.
2024-12-15 15:20:13 raspberrypi matter_server.server.server[2609] INFO Starting the Matter Server...
2024-12-15 15:20:13 raspberrypi matter_server.server.helpers.paa_certificates[2609] INFO Skip fetching certificates (already fetched within the last 24h).
2024-12-15 15:20:13 raspberrypi chip.FabricAdmin[2609] INFO Allocating new controller with CaIndex: 1, FabricId: 0x0000000000000001, NodeId: 0x000000000001B669, CatTags: []
2024-12-15 15:20:14 raspberrypi chip.native.DIS[2609] Level 39 Failed to advertise records: src/inet/UDPEndPointImplSockets.cpp:421: OS Error 0x02000065: Network is unreachable
2024-12-15 15:20:14 raspberrypi chip.native.DL[2609] Level 39 Long dispatch time: 406 ms, for event type 2
2024-12-15 15:20:14 raspberrypi matter_server.server.vendor_info[2609] INFO Loading vendor info from storage.
2024-12-15 15:20:14 raspberrypi matter_server.server.vendor_info[2609] INFO Loaded 270 vendors from storage.
2024-12-15 15:20:14 raspberrypi matter_server.server.vendor_info[2609] INFO Fetching the latest vendor info from DCL.
2024-12-15 15:20:14 raspberrypi matter_server.server.vendor_info[2609] INFO Fetched 269 vendors from DCL.
2024-12-15 15:20:14 raspberrypi matter_server.server.vendor_info[2609] INFO Saving vendor info to storage.
2024-12-15 15:20:14 raspberrypi matter_server.server.device_controller[2609] INFO Loaded 0 nodes from stored configuration
2024-12-15 15:20:14 raspberrypi matter_server.server.server[2609] INFO Matter Server successfully initialized.
2024-12-15 15:20:14 raspberrypi matter_server.client.connection[2609] INFO Connected to Matter Fabric 1 (18396262687323259652), Schema version 11, CHIP SDK Version 2024.11.4
2024-12-15 15:20:14 raspberrypi matter_server.client[2609] INFO Matter client initialized.
2024-12-15 15:20:14 raspberrypi zeroconf[2609] WARNING Error with socket 23 (('::1', 5353, 0, 0))): [Errno 101] Network is unreachable
Traceback (most recent call last):
  File "/usr/lib/python3.12/asyncio/selector_events.py", line 1275, in sendto
    self._sock.sendto(data, addr)
OSError: [Errno 101] Network is unreachable

**Comment by marcelveldt** [comment]
Didn't you read the documentation ?
Didn't you run the setup script ? Why are you manually installing the requirements ?
Sorry, this is python basic stuff.

**Comment by bkoblenz** [comment]
I thought I had read the documentation but obviously I missed something
important.  Sorry.

Rather than create a venv just for this package, I wanted to incorporate
these requirements into an existing venv.
It looked like the scripts/setup.sh would do that, but I did not get the
expected behavior.

When I looked a little closer it appears that the script has the line:
source .venv/bin/activate
when maybe it should be:
source $env_name/bin/activate

Is that an error in the script or in my thinking of how it can be used?
